smart/model_assembly.py

```diff
@@ -1787,3 +1812,43 @@ def sbmodel_from_locals(local_values):
     cc.add(compartments)
     rc.add(reactions)
     return pc, sc, cc, rc
+
+
+def create_restriction(u, mesh_function, value):
+    """
+    Restrict a function on a submesh to a subset of parent entities
+    (same dimension as the submesh)
+
+    :param u: Function on submesh
+    :param mesh_function: MeshFunction marking the subset of parent entities
+    (same dimension as the cells of the submesh)
+    :param value: Value in MeshFunction marking the subset of parent entities
+    :return: New restricted function
+    """
+    submesh = u.function_space().mesh()
+
+    # Compute local cells in submesh marked by parent meshtag
+    # using first map to run without specifying parent mesh id
+    sub_to_parent_map = list(submesh.topology().mapping().values())[0].cell_map()
```

Comment:
Here I assumed that the first map should be used, so I guess this wouldn't work if multiple maps were stored here.

Reply:
What was the rationale behind this? Do you not have access to the parent mesh?

Reply:
No access to the parent mesh here strictly speaking, could figure out a strategy to pass it in somehow.

Reply:
Hm, doesn’t the sub mesh have a pointer to the parent? I will think about this.
I guess it would be resolved if we use the id of the mesh in the meshtag, as discussed above

Reply:
Fixed now.